### ISSUE TITLE
Why not self-host the analytics.js file either?

### ISSUE THREAD

**Issue by eimanip** [scriptex/localga#7]
Hi,

I'm using gtag.js for my web application. Currently, it loads two files from the Google server. After downloading the first one, it would request for the second one. As you may already know, these files are:

- `gtag.js` file (https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXXX-Y)
- `analytics.js` file (https://www.google-analytics.com/analytics.js)

Your script only self-host the first one, while it fetches the second file from the Google server.
I am wondering whether it is possible to self-host the `analytics.js` file either? Were there any reasons behind not doing it?

Thanks in advance.

**Comment by scriptex** [comment]
Hi and thanks for the feedback. 
In order to self host this file I need to edit the source code of the original `gtag.js` file and stop it from downloading the analytics file. 
This will introduce a lot of complexity and possible errors which I don't find worth the effort.
Please share your thoughts as I might be missing something :)

**Comment by eimanip** [comment]
Thanks for your response.

As you said, it needs to update the `analytics.js` path in `gtag.js` file. But I think just self-hosting one of these two files won't improve the performance much.

Let me explain our situation to you.

We are aiming to improve our initial page load time as much as possible. As you may know, downloading these resources and initial collecting of Gtag events (i.e. page view events), have a great impact on the page load time. On the other hand, our servers are located in a country that Google does not provide any CDN for fetching these resources. As a result, it would be better for us to just self-host them on our servers and serve them from a local CDN.

Currently, I have self-hosted both of these files as a temporary solution to overcome this issue. But it needs hard-coding the tracking ID and including these files in the git repository. I'm looking for a better approach to:
- Be able to soft-code the tracking ID as before.
- Self-host both of these files on our servers.

I think one solution would be to extend your work to change the `gtag.js` contents as well at the build time, pointing to a local copy of `analytics.js` file. I would be happy to implement this feature and send you a pull request if you think it's a good idea.

Looking forward to hearing from you for your opinion about it or any alternative solution.

**Comment by scriptex** [comment]
Hi, actually the whole purpose of this module is to provide the user with the ability to self-host the analytics JavaScript file(s). 
My fear is that at some point google will change their implementation and users which use this module will experience issues.
Anyway, I am open for suggestions so feel free to tweak the implementation and open a PR.
Looking forward! :)

**Comment by scriptex** [comment]
Hello, @eimanip, is there any progress on this?

**Comment by eimanip** [comment]
Hi @scriptex ,
Actually, we are investigating to migrate to Google Tag Manager. Due to the dynamic generation of the JS file from Google, it would not be a good idea to self-host these files anymore.
If we decide to keep using the gtag.js, I will go for it for sure to dynamically generate the `analytics.js` file either.
Sorry if I kept you waiting.

P.S. We are using the self-host version of both files on production for over a month. The good news is that we did not face any issues until now. I didn't update the files in this period.

**Comment by scriptex** [comment]
Thanks for clarifying. I just wanted to know whats the status. So, correct me if I am wrong but I understand that you are not interested in this feature anymore. If yes, should I close this issue? If no, do you plan on implementing it?

**Comment by eimanip** [comment]
Yes, I currently have no plans to implement this feature in the short-term. But of course, if you implement this feature I would definitely appreciate it and use it in our project, if we keep using gtag.js.
About whether it is a needed feature or not: I honestly believe that self-hosting one of the two files isn't persuading enough to use this library in a project.
Thank you a lot for following up this issue and considering it very well.

**Comment by scriptex** [comment]
I've drafted something in https://github.com/scriptex/localga/pull/8. 
When I have more time I will try to publish it.
Thanks for the valuable feedback! :)
